src/common/snippets/src/op/subgraph.cpp

```diff
@@ -77,7 +77,42 @@ auto Subgraph::is_domain_sensitive_op(const std::shared_ptr<ov::Node>& op) -> bo
            ov::is_type<ov::op::v8::Softmax>(op) ||
            ov::is_type<ov::op::v0::MatMul>(op) ||
            ov::is_type<ov::op::v1::Broadcast>(op) || // Broadcast is domain sensetive op because the output shape depends on
-           ov::is_type<ov::op::v3::Broadcast>(op);   // the both input and broadcast shapes (the both - are inputs of op). Note: is used only in MHA pattern
+           ov::is_type<ov::op::v3::Broadcast>(op) ||   // the both input and broadcast shapes (the both - are inputs of op). Note: is used only in MHA pattern
+           ov::is_type<ov::op::v12::GroupNormalization>(op) ||
+           ov::is_type<op::Reshape>(op);
+}
+
+auto Subgraph::is_shape_infer_op(const std::shared_ptr<ov::Node>& op) -> bool {
+    return ov::is_type<snippets::op::Reshape>(op) ||
+           ov::is_type<snippets::op::RankNormalization>(op);
+}
+
+auto Subgraph::get_last_shape_infer_op(const std::shared_ptr<ov::Node>& op, bool downstream) -> std::shared_ptr<ov::Node> {
+    auto last_op = op;
+    if (downstream) {
+        if (last_op->get_output_size() == 0)
+            return last_op;
+        auto first_child = last_op->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
```

Comment:
it seems like this method must be called only for nodes with single consumer (or with single parent if `downstream==false`). Maybe it is worth adding an openvino assert which will perform the corresponding check (for both consumers and parents)? What do you think?

Reply:
At current, the shape infer ops should be single consumer from its parent, but could consumed by multiple child nodes. In these new helpers, first restriction could already guarantee it's a sequence. In GN pattern, reshape is consumed by reduceSum and substract. Thanks Vladislav!